### DISCUSSION TITLE
Changing JVT tagger via the Python config

### DISCUSSION THREAD

**Discussion by rbarrue** [UCATLAS/xAODAnaHelpers#1626] [Q&A] [answered]
Currently, the C++ code in JetSelector uses the enum defined in the JVT namespace, but that enum can't be accessed in the Python config (only via the int directly). 

One way would be reimplement the JVT tagger enum in xAH, but that would require tracking the JVT enum and change in xAH anytime it changes in the ATLAS code.

Is there a smarter way to do this ?

**Comment by jbossios** [comment, marked as answer]
EDIT: Decided to simply update the docs and the header for JetSelector to tell the user to use the corresponding index from the enum here: https://acode-browser1.usatlas.bnl.gov/lxr/source/athena/PhysicsAnalysis/Interfaces/JetAnalysisInterfaces/JetAnalysisInterfaces/IJetJvtEfficiency.h#0022
This is the simplest solution and in general people shouldn't be needing to update the JVT algorithm.

I guess this is related to your question on Mattermost. Are you talking about m_JvtTaggingAlg? I don't see the need for anyone to use any other tagger than NNJvt for offline jets. For online jets, I guess no one is actually planning to apply a JVT selection to online jets, since there is likely already a HLT jet collection where that selection was already done. If you need to apply JVT on online jets for some studies, we can find a way for you to change the tagger so that you can use JVT instead of NNJvt for online jets.

**Reply by rbarrue** [reply]
If the online jet collections are already decorated with the JVT decision derived online, we'll just want to apply NNJvt (to study its impact), so this questions becomes void.

If not, we can just give the integer mapped by the enum (keeping a link to its definition in athena code, so that it can be kept up to date). We tested it like this and it works. No point in going through a lot of work to implement something based on deprecated tools, I think.

**Reply by jbossios** [reply]
Your proposal sounds good to me. Would you like to create a PR with such a change?

**Reply by rbarrue** [reply]
This would not be in the xAH code, because being C++, it can access the enum with the member names, it would be simply a comment in the Python configs. Do you have example configs where that could be added or should I just add the comment in the header file saying "Following enum in https://gitlab.cern.ch/atlas/athena/-/blob/master/PhysicsAnalysis/Interfaces/JetAnalysisInterfaces/JetAnalysisInterfaces/IJetJvtEfficiency.h, if you need to change this in your config, use the integer mapped by the enum directly" ?

**Reply by jbossios** [reply]
I see, right, there is no need for a change in xAH. You just check that header file and pick the value that you need in your python config. I will update the docs to add information about it and edit the header file adding a comment similar to what you proposed. Thanks for the feedback.
